transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -153,7 +150,10 @@ public ChannelHandlerContext fireUserEventTriggered(Object event) {
     @Override
     public ChannelHandlerContext fireChannelRead(Object msg) {
         AbstractChannelHandlerContext next = findContextInbound();
-        ReferenceCountUtil.touch(msg, next);
+        if (ResourceLeakDetector.ENABLED) {
```

Comment:
@ninja- - Start your jvm with `-XX:+UnlockDiagnosticVMOptions -XX:+PrintInlining` to see what the jvm does. It should be small enough to be compiled (smaller than MaxInlineSize and FreqInlineSize).